spec/controllers/courses_controller_spec.rb

```diff
@@ -1,6 +1,85 @@
 describe CoursesController do
   let(:instructor) { create :instructor }
   let(:course) { instructor.course }
+
+  describe 'role switching methods' do
+    let(:end_user) { create :end_user }
```

Comment:
You don't need to define this here. Instead, use a `let` within different context blocks below to define it as an instructor/TA/student. For example, something like

```ruby
context 'when switching to the current instructor' do
  let(:end_user) { instructor }
  it ...
```

Specifically, the `let` statement for `subject` can appear *before* the `let` statement for `end_user`, even if `subject` is defined using `end_user`.